native/server/src/lib.rs

```diff
@@ -96,13 +96,13 @@ impl TryFrom<proto::service::Message> for protocol::x3dh::Message {
                 .to_vec(),
         )
         .map_err(|_| Status::invalid_argument("Invalid ciphertext."))?;
-
+    
         Ok(protocol::x3dh::Message {
             sender_identity,
             sender_identity_key,
             ephemeral_key,
             one_time_key,
-            ciphertext,
+            ciphertext: ciphertext.into_bytes(),
```

Comment:
I'm quite certain this won't work. Did you test this? The above line assumes the ciphertext is valid UTF-8 and will return an error if it is not. Removing the hex encoding steps will practically guarantee the ciphertext is not valid UTF-8. The above string decoding step should be removed.

```
let ciphertext = value.ciphertext.ok_or(Status::invalid_argument("request missing ciphertext"))?;
```